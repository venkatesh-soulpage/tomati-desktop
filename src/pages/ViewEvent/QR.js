import React from "react";

const QR = ({ event }) => {
  return (
    <div className="card bg-white border p-5 mt-2">
      <div className="d-flex align-items-center">
        <div className="mr-5">
          <img
            className="border"
            src={event && event.menu_link}
            height="204px"
            width="204px"
          />
        </div>
        <div className="mr-auto">
          <h4 className="text-dark">{event && event.name}</h4>
          <p className="text-dark font-weight-light">
            {event && event.description}
          </p>
          <button className="btn btn-danger mr-3 rounded-pill">
            Download QR Code
          </button>
          <button className="btn btn-outline-dark rounded-pill ">
            Order Menu tags
          </button>
        </div>
      </div>
    </div>
  );
};

export default QR;
