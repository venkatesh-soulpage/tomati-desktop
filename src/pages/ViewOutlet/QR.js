import React from "react";
import { downloadQr } from "utils/helper";

const QR = ({ outlet }) => {
  return (
    <div className="card bg-white border p-5 mt-2">
      <div className="d-flex align-items-top">
        <div className="mr-5">
          <img
            className="border"
            src={outlet && outlet.menu_link}
            height="204px"
            width="204px"
            id="menu-qr"
          />
        </div>
        <div className="mr-auto">
          <h4 className="text-dark">{outlet && outlet.name}</h4>
          <p className="text-dark font-weight-light w-56">
            {outlet && outlet.description}
          </p>
          <button
            className="btn btn-danger mr-3 mt-1 rounded-pill"
            onClick={() => downloadQr(outlet.name)}
          >
            Download QR Code
          </button>
          <button className="btn btn-outline-dark mt-1 px-4 rounded-pill ">
            Order Menu tags
          </button>
        </div>
      </div>
    </div>
  );
};

export default QR;
