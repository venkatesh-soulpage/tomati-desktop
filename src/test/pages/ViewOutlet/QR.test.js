import React from "react";
import { render } from "@testing-library/react";
import QR from "pages/ViewOutlet/QR";
import { Router } from "react-router-dom";
import history from "utils/history";
import { Provider } from "react-redux";
import configureStore from "store";

describe("QR page", () => {
  it("matches snapshot", () => {
    const store = configureStore();
    const { asFragment } = render(
      <Provider store={store}>
        <Router history={history}>
          <QR />
        </Router>
      </Provider>,
      {}
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
