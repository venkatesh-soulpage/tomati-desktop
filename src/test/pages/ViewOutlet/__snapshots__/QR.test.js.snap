// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QR page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card bg-white border p-5 mt-2"
  >
    <div
      class="d-flex align-items-top"
    >
      <div
        class="mr-5"
      >
        <img
          alt="menu-qr"
          class="border"
          height="204px"
          id="menu-qr"
          width="204px"
        />
      </div>
      <div
        class="mr-auto"
      >
        <h4
          class="text-dark"
        />
        <p
          class="text-dark font-weight-light w-56"
        />
        <button
          class="btn btn-danger mr-3 mt-1 rounded-pill"
        >
          Download QR Code
        </button>
        <button
          class="btn btn-outline-dark mt-1 px-4 rounded-pill "
        >
          Order Menu tags
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
